seqr/views/utils/anvil_metadata_utils.py

```diff
@@ -136,7 +138,8 @@ def _get_family_metadata(family_filter, family_fields, include_metadata, include
     family_data_by_id = {}
     for f in family_data:
         family_id = f.pop('id')
-        solve_status = ANALYSIS_SOLVE_STATUS_LOOKUP.get(f['analysisStatus'], Individual.UNSOLVED)
+        analysis_status = f['analysisStatus'] if include_metadata else f.pop('analysisStatus')
```

Comment:
Can you tell me why we have the `include_metadata` flag  in `parse_anvil_metadata() ` -  what the 'metadata' is and why we only want to return it sometimes? A lot of fields depend on it and I want to understand it!

Reply:
So this helper function is used 5 times - in 2 places to generate multi-file reports with a strict data schema (the Gregor and Anvil exports) and in 3 places to generate "metadata" reports (family, variant, and individual) that we define and use internally to help us look into our own data. These metadata reports are often close to the gregor report because one of the things we use them for ins QCing those reports, but they are also useful for other purposes and include some additional information not required by the main reports, and also they split apart/aggregate certain fields based on the primary entity for said report (i.e. some fields in the family metadata are the aggregate of that same field in the individual metadata). 

The `include_metadata` flag is True for the 3 metadata reports and False for the other 2 usages, so "metadata" here refers to the name of the reports where this additional information is valuable. 

TBH, there is probably a bunch of fields that are included when `include_metadata` is enabled thats not needed for all 3 metadata reports, and at some point I would love to split out as much of the `include_metadata` logic as possible from this one boolean flag into more targeted per-function logic, but the metadata reports are incredibly tolerant of having extra fields returned in the server response so it does not break anything for now, and I thought that was out of scope for what I wanted to accomplish here. But its definitely a future cleanup goal!

Reply:
thank you for the explanation!! From a maintainability and readability standpoint, an OOP approach might work well here with inheritance between gregor/metadata and classmethods to instantiate the reports.